akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -51,12 +51,15 @@ private[http] object HttpServerBluePrint {
     stack(settings, remoteAddress, log)
 
   def stack(settings: ServerSettings, remoteAddress: Option[InetSocketAddress], log: LoggingAdapter)(implicit mat: Materializer): Http.ServerLayer = {
-    userHandlerGuard(settings.pipeliningLimit) atop
-      requestPreparation(settings, remoteAddress) atop
-      controller(settings, log) atop
-      parsingRendering(settings, log) atop
-      websocketSupport(settings, log) atop
-      unwrapTls
+    val theStack =
+      userHandlerGuard(settings.pipeliningLimit) atop
+        requestPreparation(settings) atop
```

Comment:
That sounds somewhat more hacky and disappointing in a way, IMHO Attributes _are_ the side channel we have built for these kinds of situations. Putting it into the Flow is something impls without an infra around the streams like we have would start out with I guess.